documentation/modules/oauth/con-oauth-authentication-flow.adoc

```diff
@@ -1,12 +1,83 @@
 
-Kafka brokers and clients need to be configured to use OAuth 2.0.
+OAUTHBEARER and PLAIN can be enabled at the same time, on the same `oauth` listener, if both types of OAuth clients need to authenticate.
+
+.OAUTHBEARER overview
+
+Support for OAUTHBEARER is built in to many Kafka clients. 
+However, the developer must often write custom callbacks that provide access tokens for authentication, as needed.
+Strimzi provides these callbacks for standard Java Kafka clients (clients that use the Kafka producer and consumer APIs). 
+Therefore, developers do not need to provide custom callbacks in their application code. 
+
+With OAUTHBEARER, the client initiates a session with the Kafka broker for credentials exchange, where credentials take the form of a bearer token provided by the Strimzi OAuth callback code. 
+Using the callbacks, you can configure token provision in one of three ways:
+
+* Client ID and Secret (by using the OAuth 2.0 client credentials mechanism)
+
+* A long-lived access token, obtained manually at configuration time
+
+* A long-lived refresh token, obtained manually at configuration time
+
+OAUTHBEARER is automatically enabled in the `oauth` listener configuration for the Kafka broker. 
+You can set the `enableOauthBearer` property to `true`, though this is not required.
+
+[source,yaml,subs="attributes+"]
+----
+  # ...
+  authentication:
+    type: oauth
+    # ...
+    enableOauthBearer: true
+----
+
+.PLAIN overview
+
+PLAIN is a simple authentication mechanism supported by all Kafka client tools (including developer tools such as kafkacat). 
+To enable PLAIN to be used together with OAuth authentication, Strimzi includes additional server-side callbacks and calls this _OAuth 2.0 over PLAIN_. 
+
+With PLAIN, Client credentials are handled centrally, behind an authorization server that complies with OAuth 2.0 and OpenID Connect (OIDC).
+
+When used with the provided callbacks, PLAIN allows Kafka clients to authenticate with Kafka brokers using either of the following methods:
+
+* Client ID and Secret (by using the OAuth 2.0 client credentials mechanism)
+
+* A long-lived access token, obtained manually at configuration time
+
+The client must enable PLAIN and set the `username` and `password`. 
+If the client prefixes the `password` with `$accessToken:` followed by the value of the access token, the Kafka broker will interpret the password as the access token. 
+Otherwise, the Kafka broker will interpret the `username` as the client ID and the `password` as the client secret.
+
+If the `password` is set as the access token, the `username` must be set to the same principal name that the Kafka broker obtains from the access token. 
+The process depends on how you configure username extraction (using `userNameClaim`, `fallbackUserNameClaim`, `fallbackUsernamePrefix`, and `userInfoEndpointUri`), and also your authorization server (in particular, how it maps client IDs to account names).
```

Comment:
```suggestion
The process depends on how you configure username extraction (using `userNameClaim`, `fallbackUserNameClaim`, `fallbackUsernamePrefix`, or `userInfoEndpointUri`), and also your authorization server (in particular, how it maps client IDs to account names).
```

Reply:
This is a long sentence. Maybe it could be split and the brackets brought into main sentences.